File: adress-service/src/main/java/br/com/adressservice/infrastructure/repository/ValidationRepository.java
package br.com.adressservice.infrastructure.repository;

import br.com.adressservice.core.model.Validation;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ValidationRepository extends JpaRepository<Validation, Integer> {

    Boolean existsByUserIdAndTransactionId(String userId, String transactionId);
    Optional<Validation> findByUserIdAndTransactionId(String userId, String transactionId);
}

File: user-registration/src/main/kotlin/br/com/userregistration/infrastructure/repository/ValidationRepository.kt
package br.com.userregistration.infrastructure.repository

import br.com.userregistration.core.model.Validation
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
    interface ValidationRepository : JpaRepository<Validation, Int> {
    fun existsByUserIdAndTransactionId(userId: String, transactionId: String): Boolean
    fun findByUserIdAndTransactionId(userId: String, transactionId: String): Optional<Validation>
}